### ISSUE TITLE
Invalid response code: 190. Can't add SwitchBot Smart Lock to Homey

### ISSUE THREAD

**Issue by Francas555** [AdyRock/com.switchbot#44]
Hello,
Old Remote working. But if i want add Smart Lock or Bot i get error 190.
O have two hubs, one at office (with Smart Lock), other at Home.

**Comment by AdyRock** [comment]
From the Switchbot document:
```
Error 190
Device internal error due to device states not synchronized with server. Or command format is invalid.
```
My lock is working fine, so I don't think the command format is an issue. 
Maybe try again later just in case there is a server issue and get back to me.

**Comment by Francas555** [comment]
thanks, I will try tomorrow.

**Comment by Francas555** [comment]
Hi Adrian :)
I did some experiments.
Maybe Google account doesn't work with Homey? I can't add new devices, but the old ones working.
I found that I have an account with my email and with Google.
My email account shows Bot for adding to Homey, and Gmail shows error 190.

**Comment by Francas555** [comment]
Hi Adrian :)
Problem solved for me. I move Lock to my email account and add to Homey.
But why not working from google Account?

**Comment by AdyRock** [comment]
Great, I'm glad it's working. 
I'm not sure about the account discrepancy. I presume Homey was also using the Google account?
I use my email account and haven't tried with a Google account. 
When you log into the Switchbot phone app, can you use either method and see the devices?

**Comment by Francas555** [comment]
That's the problem.
I see different devices with my email and with my Google account - like different Accounts.
But my account is connected to Google (it seems so to me).
But it seems to me that switchBot cannot be used only with a Google Account, new user need to sign up with own email first!?

**Comment by Francas555** [comment]
I checked, new user with Google Account is not allowed.

**Comment by Guinnex** [comment]
Hello, I have the same error when I log in with my username and password. I can't use login with my google account. (my switchbot account was created with my gmail address). On the switchbot connection applet, I am well connected with my account but when I return to homey, I have the error: invalid response code 190.
thank you
Guillaume MONJEAN

**Comment by Francas555** [comment]
Hello, The weekend was fine connect to Homey with my email.
Not working yesterday and today, error code 190.
I have two Locks, 2 Hubs. Maybe this is problem?

**Comment by Guinnex** [comment]
I only have one hub and one lock. I put a nickname on the switchbot app and I can only log in with the nickname now. I can no longer write my gmail address. I don't know if this is a problem...

**Comment by xtiancma** [comment]
1 hub, using only email account (no gmail), cant add any switchboot devices - getting the same 190 error

**Comment by Robbedush** [comment]
I am having the same issue trying to add the Lock Pro to my Homey Pro 2023. I keep getting the invalid respond 190 error. I talked to Switchbot Support and they said it's not an error in their system and I should contact Homey Support. 

Both my Homey or Switchbot accounts are not registered to Google/Gmail. 

I will try to contact Homey Support but would anyone have an idea what could be the cause? I have a Switchbot Hub Mini Matter, a keypad and the Lock Pro.

**Comment by AdyRock** [comment]
Could you open the SwitchBot phone app, go to Profile, Preferences, About and then keep tapping on the version number (at least 11 times). At some point a new item should appear called Developer Options.
Once you have done that, try to add the lock again.

**Comment by Robbedush** [comment]
> Could you open the SwitchBot phone app, go to Profile, Preferences, About and then keep tapping on the version number (at least 11 times). At some point a new item should appear called Developer Options. Once you have done that, try to add the lock again.

Yeah, I just found that about an hour ago. Absolutely great! Thanks for that. 

A little bit crazy that Switchbot Support doesn't know this. I sent them the reply with a link to Github so hopefully that will fix that. And also asked if they could add Deadbolt Unlock only instead of opening the door as well.
